src/vs/workbench/browser/parts/compositebar/compositeBarActions.ts

```diff
@@ -207,7 +207,15 @@ export class ActivityActionItem extends BaseActionItem {
 			// Number
 			if (badge instanceof NumberBadge) {
 				if (badge.number) {
-					this.$badgeContent.text(badge.number > 99 ? '99+' : badge.number.toString());
+					let number;
+					if (badge.number > 9999) {
+						number = '10k+';
+					} else if (badge.number > 999) {
+						number = (badge.number / 1000).toFixed(1).replace(/\.0$/, '') + 'k';
```

Comment:
IMHO this just seems like too complex of a logic. 
Can we use something like 

```
badge.number.toString()[0] + 'k'
```

Reply:
Ah yes! Because we know the number is < 10,000, we can do that 👍

Reply:
Going to use `.charAt(0)` as it's more readable.